packages/flutter_tools/test/integration.shard/xcode_backend_test.dart

```diff
@@ -132,27 +132,47 @@ void main() {
     });
 
     for (final buildConfiguration in <String>['Debug', 'Profile']) {
-      test('add keys in $buildConfiguration', () async {
-        infoPlist.writeAsStringSync(emptyPlist);
-
-        final ProcessResult result = await Process.run(
-          xcodeBackendPath,
-          <String>['test_vm_service_bonjour_service'],
-          environment: <String, String>{
-            'CONFIGURATION': buildConfiguration,
-            'BUILT_PRODUCTS_DIR': buildDirectory.path,
-            'INFOPLIST_PATH': 'Info.plist',
+      for (final verbose in <bool>[true, false]) {
+        test(
+          'add keys in $buildConfiguration under ${verbose ? 'verbose' : 'non-verbose'} mode',
+          () async {
+            infoPlist.writeAsStringSync(emptyPlist);
+            final File pipe = fileSystem.file('/tmp/pipe')..createSync(recursive: true);
+
+            final ProcessResult result = await Process.run(
+              xcodeBackendPath,
+              <String>['test_vm_service_bonjour_service'],
+              environment: <String, String>{
+                'CONFIGURATION': buildConfiguration,
+                'BUILT_PRODUCTS_DIR': buildDirectory.path,
+                'INFOPLIST_PATH': 'Info.plist',
+                if (verbose) 'VERBOSE_SCRIPT_LOGGING': 'YES',
+                'SCRIPT_OUTPUT_STREAM_FILE': pipe.path,
+              },
+            );
+
+            final String actualInfoPlist = infoPlist.readAsStringSync();
+            expect(actualInfoPlist, contains('NSBonjourServices'));
+            expect(actualInfoPlist, contains('dartVmService'));
+            expect(actualInfoPlist, contains('NSLocalNetworkUsageDescription'));
+
+            // Make sure no Xcode compilation error.
+            expect(result.stderr, isNot(startsWith('error:')));
+
+            const plutilErrorMessage =
+                'Could not extract value, error: No value at that key path or invalid key path: NSBonjourServices';
+            expect(pipe.readAsStringSync(), isNot(contains(plutilErrorMessage)));
+            expect(result.stderr, isNot(contains(plutilErrorMessage)));
+            if (verbose) {
+              expect(result.stdout, contains(plutilErrorMessage));
```

Comment:
verified on both macOS 26 and 15.6: 

On macOS 26:
  - if verbose, plutil logs to its stderr, and we redirect to stdout
  - if non-verbose, plutil logs to its stderr, and we ignore it

On macOS 15.6
  - if verbose, plutil logs to its stdout, and we simply log it to current process's stdout
  - if non-verbose, plutil logs to its stdout and we ignore it.